lib/src/models/function_type.dart

```diff
@@ -0,0 +1,26 @@
+/// Enum class for type of a function entity.
+///
+/// Used when reporting.
+class FunctionType {
+  /// The class entity representing a class constructor.
+  static const constructor = FunctionType._('constructor');
+
+  /// The class entity representing a class method.
+  static const method = FunctionType._('method');
+
+  /// The class entity representing a function.
+  static const function = FunctionType._('function');
```

Comment:
I think the comment should not mention a class then if we talk about a function?

Reply:
Or is it about an inner function inside of a method?